### PR TITLE
Add WebSocketHandler code in main middleware

### PR DESCRIPTION
**Pull request by Shane32**
Alternative design to #812 

Todo: create PR in GraphQL.AspNetCore3 to match with updated tests to ensure everything works correctly

### CODE REVIEW

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs [hunk 1]
```diff
@@ -157,18 +109,50 @@ protected override async Task<ExecutionResult> ExecuteRequestAsync(HttpContext c
...
 
 /// <summary>
 /// ASP.NET Core middleware for processing GraphQL requests. Handles both single and batch requests,
 /// and dispatches WebSocket requests to the registered <see cref="IWebSocketHandler"/>.
 /// </summary>
-public abstract class GraphQLHttpMiddleware
+public abstract class GraphQLHttpMiddleware : IUserContextBuilder
 {
     private readonly IGraphQLTextSerializer _serializer;
-    private readonly IEnumerable<IWebSocketHandler>? _webSocketHandlers;
```
Comment (sungam3r) [inline src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:169]: I see warnings about `IWebSocketHandler` in xml comments.

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs [hunk 2]
```diff
@@ -63,24 +43,8 @@ public class GraphQLHttpMiddleware<TSchema> : GraphQLHttpMiddleware
...
-    /// </summary>
-    protected GraphQLHttpMiddleware(
-        RequestDelegate next,
-        IGraphQLTextSerializer serializer,
-        IDocumentExecuter<TSchema> documentExecuter,
-        IServiceScopeFactory serviceScopeFactory,
-        GraphQLHttpMiddlewareOptions options,
-        IEnumerable<IWebSocketHandler<TSchema>>? webSocketHandlers = null)
-        : base(next, serializer, options, webSocketHandlers)
+        : base(next, serializer, options, hostApplicationLifetime)
```
Comment (sungam3r) [inline src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:47]: Technically `hostApplicationLifetime` _may_ be optional. At least this is nothing more than an additional source of information about cancellation.

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs (hunk 1 above)
Comment (Shane32) [reply src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:169]: Fixed

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs (hunk 2 above)
Comment (sungam3r) [reply src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:47]: Although I forget (again) that the project is called Transports.**AspNetCore** so there is always should be `hostApplicationLifetime`.

**Pull request merged**